tests/integration/plugins/preprocess/plugin.cjs

```diff
@@ -11,14 +11,14 @@ module.exports = {
   ],
   parsers: {
     "foo-parser": {
-      preprocess: (text) => `preprocessed:${text}`,
+      preprocess: (text) => Promise.resolve(`preprocessed:${text}`),
```

Comment:
If we don't have a test for non-promise besides this one, I think we should add a new test instead of modifying this one.

Reply:
[`0d3bae1` (#17679)](https://github.com/prettier/prettier/pull/17679/commits/0d3bae1b2ad5b46b07af65d93b6bb279324021a0)